frontend/packages/console-dynamic-plugin-sdk/src/coderefs/coderef-resolver.ts

```diff
@@ -126,3 +126,33 @@ export const resolveCodeRefProperties = async <E extends Extension<P>, P = Exten
 
   return resolvedValues;
 };
+
+/**
+ *  Returns an extension with its resolved `CodeRef` properties.
+ */
+export const resolveExtensionWithCodeRefs = <E extends Extension<P>, P = ExtensionProperties<E>>(
+  ext: E,
+): Promise<Extension<ResolvedCodeRefProperties<P>>> => {
+  const refProperties = filterExecutableCodeRefProperties(ext.properties);
+  const refPromises = Object.values(refProperties).map((ref) => ref());
+
+  return Promise.all(refPromises).then((properties) => {
+    const resolvedRefs: ResolvedCodeRefProperties<P> = Object.assign(
+      {},
+      ...Object.keys(refProperties).map((key, i) => ({
+        [key]: properties[i],
+      })),
+    );
+
+    return Object.freeze(_.merge({}, ext, { properties: { ...resolvedRefs } }));
+  });
+};
+
+/**
+ *  Returns an array extensions with their resolved `CodeRef` properties.
+ */
+export const resolveExtensionsArray = <E extends Extension<P>, P = ExtensionProperties<E>>(
+  extensions: E[],
+): Promise<Extension<ResolvedCodeRefProperties<P>>[]> => {
```

Comment:
```suggestion
): Promise<ResolvedExtension<E>[]> => {
```

Reply:
This doesn't make any sense..the extension isn't resolved yet, why put it inside a promise that is not resolved?
@vojtechszocs